packages/core/upload/admin/src/hooks/tests/useAssets.test.js

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable import/order */
+/* eslint-disable import/first */
+
+import React from 'react';
+import { IntlProvider } from 'react-intl';
+import { QueryClientProvider, QueryClient } from 'react-query';
+import { renderHook } from '@testing-library/react-hooks';
+import { BrowserRouter as Router, Route } from 'react-router-dom';
+
+import { NotificationsProvider } from '@strapi/helper-plugin';
+
+import { useAssets } from '../useAssets';
+
+const notifyStatusMock = jest.fn();
+
+jest.mock('@strapi/design-system/LiveRegions', () => ({
+  ...jest.requireActual('@strapi/design-system/LiveRegions'),
+  useNotifyAT: () => ({
+    notifyStatus: notifyStatusMock,
+  }),
+}));
+
+import { useNotifyAT } from '@strapi/design-system/LiveRegions';
+
+jest.mock('../../utils', () => ({
+  ...jest.requireActual('../../utils'),
+  axiosInstance: {
+    get: jest.fn().mockResolvedValue({
+      data: {
+        id: 1,
+      },
+    }),
+  },
+}));
+
+import { axiosInstance } from '../../utils';
+
+// eslint-disable-next-line react/prop-types
+function ComponentFixture({ children }) {
+  return (
+    <Router>
+      <Route>
+        <QueryClientProvider client={new QueryClient()}>
+          <NotificationsProvider toggleNotification={() => jest.fn()}>
+            <IntlProvider locale="en" messages={{}}>
+              {children}
+            </IntlProvider>
+          </NotificationsProvider>
+        </QueryClientProvider>
+      </Route>
+    </Router>
+  );
+}
+
+function setup(...args) {
+  return renderHook(() => useAssets(...args), { wrapper: ComponentFixture });
+}
+
+describe('useAssets', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('fetches data from the right URL', async () => {
+    const { result, waitFor } = setup({});
+
+    await waitFor(() => result.current.isSuccess);
+
+    expect(axiosInstance.get).toBeCalledWith('/upload/files');
+  });
+
+  test('it does not fetch, if skipWhen is set', async () => {
+    const { result, waitFor } = setup({ skipWhen: true });
+
+    await waitFor(() => result.current.isSuccess);
+
+    expect(axiosInstance.get).toBeCalledTimes(0);
+  });
+
+  test('calls notifyStatus', async () => {
+    const { notifyStatus } = useNotifyAT();
+    const { result, waitFor } = setup({});
+
+    await waitFor(() => result.current.isSuccess);
+
+    expect(notifyStatus).toBeCalledWith('The assets have finished loading.');
+  });
```

Comment:
@soupette Do you have an idea, of how I could make this test work with the `useEffect` side effect? 

https://github.com/strapi/strapi/blob/6df5d8d11fa9e57aab29e50c54587c68dea436f8/packages/core/upload/admin/src/hooks/useAssets.js#L27-L36